diff --git a/Source/LibPNG/pngpriv.h b/Source/LibPNG/pngpriv.h
index 81f8761..9f0431c 100644
--- a/Source/LibPNG/pngpriv.h
+++ b/Source/LibPNG/pngpriv.h
@@ -515,7 +515,7 @@
 #  include <float.h>
 
 #  if (defined(__MWERKS__) && defined(macintosh)) || defined(applec) || \
-    defined(THINK_C) || defined(__SC__) || defined(TARGET_OS_MAC)
+    defined(THINK_C) || defined(__SC__) || (defined(TARGET_OS_MAC) && !defined(__MACH__))
    /* We need to check that <math.h> hasn't already been included earlier
     * as it seems it doesn't agree with <fp.h>, yet we should really use
     * <fp.h> if possible.
